sympy/polys/domains/complexfield.py

```diff
@@ -69,6 +69,10 @@ def from_sympy(self, expr):
         real, imag = number.as_real_imag()
 
         if real.is_Number and imag.is_Number:
+            if number is oo:
+                return inf+0.0j
```

Comment:
I think that infinities and nan should raise an exception here, and in `RR.from_sympy` as well. CoercionFailed would be the proper choice.

Reply:
It is probably not necessary to make the checks in `from_sympy`. It would suffice that ComplexElement should raise CoercionFailed.